website/src/controllers/api-v3/tasks.js

```diff
@@ -15,80 +16,220 @@ import { preenHistory } from '../../../../common/script/api-v3/preenHistory';
 
 let api = {};
 
+// challenge must be passed only when a challenge task is being created
+async function _createTasks (req, res, user, challenge) {
```

Comment:
In fact, you don't even need to pass the user in, since it's on the `res.locals.user`. You could just have the arguments be `req, res, challenge`